File: Lab-1/Task_1-C/ReverseRecursive.c
/*******************************************************************************
*
* Seminar 1 - task 1
* Author: Alexander Lundqvist
* Created: 30-08-2021
*
* Single file program that performs the task of taking in a string input and
* then print out the characters in reverse order by using a recursive function.
*
*******************************************************************************/

#include <stdio.h>
#include <stdlib.h>

void recursive() {
  char c;
  c = getchar(); // Reads a char from standard input, each time it is called it moves to the next char
  if (c != '\n') { // Checks if we reached "end of string" by checking for newline
    recursive();
    putchar(c); // Outputs a character to the standard output
  }
}

int main(void){
  printf("Input text: \n");
  recursive();
  return 0;
}

File: Lab-1/src/main/java/Task_2/ReverseRecursive.java
package Task_2;

/*******************************************************************************
*
* Seminar 1 - task 2
* @author Alexander Lundqvist
* Created: 30-08-2021
*
* Single file program that performs the task of taking in a string input and
* then print out the characters in reverse order by using a recursive function.
* 
*
*******************************************************************************/

import java.util.*;

public class ReverseRecursive {
    
    public static void recursive(char[] textArray, int index) {
        if (index >= 0){ // Checks if we come to firts element in array
            System.out.print(textArray[index--]);
            recursive(textArray, index);
        }
    }
    
    /**
     * Main method. Controls the program flow.
     * @param args takes no argument
     */
    public static void main(String[] args){
        
        Scanner input = new Scanner(System.in);
        System.out.println("Please input text to reverse: ");
        String text = input.next();
        char[] textArray = text.toCharArray();
        System.out.println();
        recursive(textArray, textArray.length-1);
    }
}
